### PR TITLE
[process-agent] add collection of init containers

### PR DESCRIPTION
**Pull request by nammn**
### What does this PR do?

- [enhancement] Addition of the collection of the init containers.
- [dep update] Bump Agent-payload to support the collecting of the init containers: https://github.com/DataDog/agent-payload/pull/55.

### Motivation

Related Jira Task: CAP-60
Collection of init containers for future work.

### Additional Notes

Anything else we should know when reviewing?

### Describe your test plan

Write there any instructions and details you may have to test your PR.

### CODE REVIEW

File: pkg/process/util/orchestrator/pod.go [hunk 1]
```diff
@@ -174,32 +174,42 @@ func extractPodMessage(p *v1.Pod) *model.Pod {
...
-			cStatus.State = "Terminated"
-			exitString := "(exit: " + strconv.Itoa(int(cs.State.Terminated.ExitCode)) + ")"
-			cStatus.Message = cs.State.Terminated.Reason + " " + cs.State.Terminated.Message + " " + exitString
-		}
+		cStatus := convertContainerStatus(cs)
 		podModel.ContainerStatuses = append(podModel.ContainerStatuses, &cStatus)
 	}
 
+	for _, cs := range p.Status.InitContainerStatuses {
+		podModel.RestartCount += cs.RestartCount
```
Comment (nammn) [inline pkg/process/util/orchestrator/pod.go:182]: Do we want to count the initContainers and their restarts as well?

Comment (nammn) [reply pkg/process/util/orchestrator/pod.go:182]: Building upon the discussion with Hassaim I will doublecheck how kubectl is doing it.

Comment (nammn) [reply pkg/process/util/orchestrator/pod.go:182]: After doublechecking it seems to be correct to add the restartCount to the overall restartCount as this models `kubectl` well.

See here:
```
vagrant@agent-dev-nam:~/go/src/github.com/DataDog/datadog-agent$ k describe pod init-demo

Init Containers:
  install:
    Container ID:  docker://b4062db0bf18b844eaded9c947754a741e7b39531a220024931d1f59960f307d
....
      Finished:     Wed, 24 Jun 2020 08:42:52 +0000
    Ready:          False
    Restart Count:  4
    
vagrant@agent-dev-nam:~/go/src/github.com/DataDog/datadog-agent$ k get pods
NAME                            READY   STATUS                   RESTARTS   AGE
init-demo                       0/1     Init:RunContainerError   5          3m12s
```

File: pkg/process/util/orchestrator/pod_test.go [hunk 2]
```diff
@@ -219,6 +219,96 @@ func TestExtractPodMessage(t *testing.T) {
...
+				Metadata: &model.Metadata{
+					Name:      "pod",
+					Namespace: "namespace",
+					OwnerReferences: []*model.OwnerReference{
+						{
+							Uid: "1234567890",
+						},
+					},
+				},
+				RestartCount: 12,
```
Comment (nammn) [inline pkg/process/util/orchestrator/pod_test.go:284]: 10 + 2 -> testing that the restartCounts are aggregated.

File: pkg/process/util/orchestrator/pod.go (hunk 1 above)
Comment (hkaj) [reply pkg/process/util/orchestrator/pod.go:182]: excellent 👌

**Review by hkaj — APPROVED** [review]

**Pull request merged**